nio_cli/commands/blockcheck.py

```diff
@@ -0,0 +1,241 @@
+                    if 'class ' in line:
+                        split1 = line.split(' ')[1]
+                        class_name = split1.split('(')[0]
+                        name_version_dict['classes'].append(class_name)
+                    if '= VersionProperty' in line:
+                        replace1 = line.replace('"', '^')
+                        replace2 = replace1.replace("'", '^')
+                        version_string = replace2.split("^")[1]
+                        name_version_dict['versions'].append(version_string[:3])
+        return name_version_dict
+
+    def run(self):
+
+        self.print_check('PEP8')
+        self.check_pep8()
+
+        self.print_check('spec.json')
+        self.check_spec()
+
+        self.print_check('README.md')
+        self.check_readme()
+
+        self.print_check('release.json')
+        self.check_release()
+
+        self.print_check('version')
+        self.check_version()
+
+        self.print_check('class and file name')
+        self.check_naming()
+
+    def check_pep8(self):
+
+        shell_pep8 = 'pep8 .'
+        subprocess.call(shell_pep8, shell=True)
+        print('')
+
+    def check_spec(self):
+
+        if os.path.exists('spec.json'):
+            with open('spec.json') as f:
+                try:
+                    spec_dict = json.load(f)
+                except json.JSONDecodeError:
+                    print('spec.json file is either incomplete or '
+                          'has an invalid JSON object')
+                    print(
+                        '\n**Run `nio buildspec {}` '
+                        'from the project directory '
+                        'and re-run this check**\n'.format(self._block)
+                    )
+                    sys.exit()
+                self.blocks_in_spec = [
+                    k.split('/')[1] for k, v in spec_dict.items()
+                ]
+
+            if len(self.blocks_in_spec) > len(self.block_files):
+                print('There are extra blocks in the spec file')
+                print(
+                    '\n**Run `nio buildspec {}` from the project directory '
+                    'and re-run this check**\n'.format(self._block)
+                )
+                sys.exit()
+            if len(self.blocks_in_spec) < len(self.block_files):
+                print('Not all blocks are in the spec file')
+                print(
+                    '\n**Run `nio buildspec {}` from the project directory '
+                    'and re-run this check**\n'.format(self._block)
+                )
+                sys.exit()
+
+            for block in self.blocks_in_spec:
+                self.spec_versions_dict[block] = \
+                    spec_dict['nio/' + block]['version']
+                keys = ['version', 'description', 'properties', 'inputs',
+                        'outputs', 'commands']
+                for key in keys:
+                    if key not in [k for k in spec_dict['nio/' + block]]:
```

Comment:
I thought we had decided that some of these are optional in the spec file? We can check with Ben to make sure.